### PR TITLE
Run `isort` and `pyupgrade` in CI

### PR DESCRIPTION
**Pull request by crusaderky**

### CODE REVIEW

**Comment by crusaderky** [comment]
@jrbourbeau after this PR _and after the next distributed release_ I think it could be a good idea to make a release

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+      rev: 5.10.1
+      hooks:
+      - id: isort
+        language_version: python3
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py38-plus
```
Comment (jrbourbeau) [inline .pre-commit-config.yaml:12]: Since we still support Python 3.7 today

```suggestion
          - --py37-plus
```

File: setup.cfg [hunk 2]
```diff
@@ -15,4 +15,12 @@ ignore =
 max-line-length = 88
 
 [tool:pytest]
-addopts = -v --doctest-modules
\ No newline at end of file
+addopts = -v --doctest-modules
+
+[isort]
+sections = FUTURE,STDLIB,THIRDPARTY,DISTRIBUTED,FIRSTPARTY,LOCALFOLDER
```
Comment (jrbourbeau) [inline setup.cfg:21]:
```suggestion
sections = FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @crusaderky  -- this seems sensible to me

**Review by jrbourbeau — APPROVED** [review]
Thanks @crusaderky -- this is in

**Pull request merged**